src/Runtime/__mem.c

```diff
@@ -7,12 +7,24 @@ void* memset(void* dst, int val, unsigned long /*size_t*/ n)
     return (dst);
 }
 
+#ifdef __MWERKS__
+#define INCREMENT_ASSIGN(ptr, type, value) (*++((type*) (ptr)) = (value))
+#else
+#define INCREMENT_ASSIGN(ptr, type, value)                                     \
+    do {                                                                       \
+        type* __INCREMENT_ASSIGN_tmp;                                          \
+        (ptr) = ((type*) (ptr)) + 1;                                           \
+        __INCREMENT_ASSIGN_tmp = ((type*) (ptr));                              \
+        *__INCREMENT_ASSIGN_tmp = (value);                                     \
+    } while (false);
+#endif
```

Comment:
This implementation was tested by substituting it on line 36, and the dol matched. Using the same macro does not work in the loop below (line 51), so keeping the original code was necessary.

If you have a more elegant solution, I'd be happy to hear it.